src/VkLayer_GF_amber_scoop/src/amber_scoop_layer.cc

```diff
@@ -42,17 +50,28 @@ struct InstanceData {
 };
 
 struct DeviceData {
-  VkDevice device;
-  InstanceData* instance_data;
+  VkDevice device = {};
+  InstanceData* instance_data = {};
 
   // Most layers must store this. Required to implement vkGetDeviceProcAddr.
   // Should not be used otherwise; all required device function pointers
   // should be obtained during vkCreateDevice.
-  PFN_vkGetDeviceProcAddr vkGetDeviceProcAddr;
+  PFN_vkGetDeviceProcAddr vkGetDeviceProcAddr = {};
 
   // Other device functions:
 
-  PFN_vkQueueSubmit vkQueueSubmit;
+  PFN_vkQueueSubmit vkQueueSubmit = {};
+  PFN_vkCmdBeginRenderPass vkCmdBeginRenderPass = {};
+  PFN_vkCmdDraw vkCmdDraw = {};
+  PFN_vkCmdDrawIndexed vkCmdDrawIndexed = {};
+  PFN_vkCmdBindPipeline vkCmdBindPipeline = {};
+
+  // Tracked device data:
+
+  std::shared_ptr<
+      ProtectedMap<VkCommandBuffer, std::vector<std::shared_ptr<Cmd>>>>
+      command_buffers = std::make_shared<
+          ProtectedMap<VkCommandBuffer, std::vector<std::shared_ptr<Cmd>>>>();
```

Comment:
Oops, that is definitely a copy-paste error.

Reply:
@paulthomson Actually, looks like it has a reason to be a `shared_ptr`. If it's not a pointer, then the copy constructor of the `DeviceData` struct will be implicitly deleted, because the mutex used in the `ProtectedMap` has disabled copy constructor. And the copy constructor is used in `vkCreateDevice` function when updating the device map: `GetGlobalData()->device_map.put(device_key(*pDevice), device_data);` So the shared pointer is just for easier memory management. Of course we could use normal pointer and implement destructor to free the memory in the `DeviceData` struct or maybe even intercept `vkDestroyDevice` to properly free the `ProtectedMap` and delete the device from the `device_map` field.

Reply:
I have a commit that I can push to fix this. But I hesitate to push it in case you have unpushed changes.